leaderelection/context.go

```diff
@@ -168,9 +160,27 @@ func (b *standardBuilder) buildElector(ctx context.Context, la reconciler.Leader
 		// if lec.WatchDog != nil {
 		// 	lec.WatchDog.SetLeaderElection(le)
 		// }
-		buckets = append(buckets, &runUntilCancelled{Elector: le})
+		electors = append(electors, &runUntilCancelled{Elector: le})
+	}
+	return &runAll{les: electors}, nil
+}
+
+func newStandardBuckets(queueName string, cc ComponentConfig) []reconciler.Bucket {
+	names := make(sets.String, cc.Buckets)
+	for i := uint32(0); i < cc.Buckets; i++ {
+		names.Insert(standardBucketName(i, queueName, cc))
 	}
-	return &runAll{les: buckets}, nil
+	bs := hash.NewBucketSet(names)
+
+	bkts := make([]reconciler.Bucket, 0, cc.Buckets)
+	for name := range names {
+		bkts = append(bkts, hash.NewBucket(name, bs))
+	}
+	return bkts
+}
+
+func standardBucketName(ordinal uint32, queueName string, cc ComponentConfig) string {
+	return strings.ToLower(fmt.Sprintf("%s.%s.%02d-of-%02d", cc.Component, queueName, ordinal, cc.Buckets))
 }
```

Comment:
This will not work well if the number of buckets were to change dynamically. I don't think we're planning for that, but should be called out in a comment just in case.

Reply:
I don't think dynamical change of bucket size is supported now. Not only this func, the whole stuff doesn't work well in bucket size change case. I don't think we need call out for this func. Instead we can call out in the user-guide or something similar for customers.

Reply:
Add a comment to `config.go` around lines 78 and 111 that changing `Buckets` dynamically is not supported?